packages/editor/src/store/reducer.js

```diff
@@ -273,6 +300,48 @@ const withBlockReset = ( reducer ) => ( state, action ) => {
 	return reducer( state, action );
 };
 
+/**
+ * Higher-order reducer which targets the combined blocks reducer and handles
+ * the `SAVE_REUSABLE_BLOCK_SUCCESS` action.
+ *
+ * @param {Function} reducer Original reducer function.
+ *
+ * @return {Function} Enhanced reducer function.
+ */
+const withSaveReusableBlock = ( reducer ) => ( state, action ) => {
+	let newState = state;
+
+	if ( state && action.type === 'SAVE_REUSABLE_BLOCK_SUCCESS' ) {
+		const { id, updatedId } = action;
+
+		// If a temporary reusable block is saved, we swap the temporary id with the final one
+		if ( id === updatedId ) {
+			return state;
+		}
+
+		newState = { ...state };
+		newState.byClientId = {};
+		newState.attributesByClientId = {};
+
+		Object.keys( state.byClientId ).forEach( ( blockId ) => {
+			const block = { ...state.byClientId[ blockId ] };
```

Comment:
Why do we create shallow clones?

Reply:
That's a good question. This must have stayed behind as I was figuring out the reducer. Sorry about that. Fixed!